### PR TITLE
OP-986: Implementation of component ValidatedTextInput

### PR DESCRIPTION
**Pull request by olewandowski1**
**PART OF [*LOCATION PR*](https://github.com/openimis/openimis-fe-location_js/pull/35)**

[OP-986](https://openimis.atlassian.net/browse/OP-986)

In scope of this ticket, I implemented new, generic component ValidatedTextInput, which checks the uniqueness of entered data and gives the appropriate information about it.

[OP-986]: https://openimis.atlassian.net/browse/OP-986?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/inputs/ValidatedTextInput.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            {!isValidating && !isValid && value && <ErrorOutlineOutlinedIcon size={20} />}
+          </>
+        </InputAdornment>
+      }
+      onChange={onChange}
+    />
+  );
+};
+
+export default UniqueValueValidation;
```
Comment (sniedzielski) [inline src/components/inputs/ValidatedTextInput.js:83]: I think the naming of file should be aligned with the name of this compoment - 'UniqueValueValidation' ->'ValidatedTextInput'.
Please use one of them: either 'UniqueValueValidation' or 'ValidatedTextInput'. @olewandowski1

File: src/components/inputs/ValidatedTextInput.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import { debounce } from "lodash";
```
Comment (sniedzielski) [inline src/components/inputs/ValidatedTextInput.js:1]: You could try to sort imports according to the rules of importing in React. We could apply those rules in new components like this. But this is not obligatory to fix this in this PR. This is only a proposal.
In my opinion we should try to remember about this while adding a whole new files into the project. 
@olewandowski1

**Review by sniedzielski — COMMENTED** [review]
I left three comments, please refer to them @olewandowski1

File: src/components/inputs/ValidatedTextInput.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import { InputAdornment, CircularProgress, Box } from "@material-ui/core";
+import { makeStyles } from "@material-ui/styles";
+import { TextInput, useModulesManager, useTranslations } from "@openimis/fe-core";
+import { useDispatch } from "react-redux";
+import CheckOutlinedIcon from "@material-ui/icons/CheckOutlined";
+import clsx from "clsx";
+import ErrorOutlineOutlinedIcon from "@material-ui/icons/ErrorOutlineOutlined";
+import React, { useEffect, useRef } from "react";
+
+const useStyles = makeStyles((theme) => ({
```
[the remark this answers is not in the document]
Comment (olewandowski1) [reply src/components/inputs/ValidatedTextInput.js]: Done, I created new file `styles.js` and moved `useStyle` there.

File: src/components/inputs/ValidatedTextInput.js (hunk 1 above)
Comment (olewandowski1) [reply src/components/inputs/ValidatedTextInput.js]: Done, it was done by oversight. Thanks for pointing this out to me.

File: src/components/inputs/ValidatedTextInput.js (hunk 2 above)
Comment (olewandowski1) [reply src/components/inputs/ValidatedTextInput.js]: I agree, it is definitely worth doing. I sorted imports according to the rules which you sent me on DMs.

**Comment by olewandowski1** [comment]
@sniedzielski I refered to your comments, could you please take a look again?

File: src/components/inputs/ValidatedTextInput.js (hunk 2 above)
Comment (sniedzielski) [reply src/components/inputs/ValidatedTextInput.js]: thank you for improving the order of imports!

**Pull request merged**